editor/src/core/model/element-metadata.spec.browser2.tsx

```diff
@@ -266,6 +266,101 @@ describe('globalContentBoxForChildren calculation', () => {
     })
   })
 
+  describe('globalcontentbox for children of sizeless divs', () => {
+    it(`globalContentBoxForChildren is correct for an absolute child of a sizeless div (with only height)`, async () => {
+      const editor = await renderTestEditorWithCode(
+        makeTestProjectCodeWithSnippet(`
+        <div
+          style={{
+            height: '100%',
+            width: '100%',
+            contain: 'layout',
+          }}
+          data-uid='a7b'
+        >
+          <div
+            style={{
+              height: 150,
+              position: 'absolute',
+              left: 140,
+              top: 130,
+            }}
+            data-uid='b15'
+          >
+            <img
+              src='https://github.com/concrete-utopia/utopia/blob/master/editor/resources/editor/pyramid_fullsize@2x.jpg?raw=true'
+              alt='Utopia logo'
+              style={{
+                height: 150,
+                width: 120,
+                left: 10,
+                top: 15,
+                position: 'absolute',
+              }}
+              data-uid='b0e'
+            />
+          </div>
+        </div>
+        `),
+        'await-first-dom-report',
+      )
+
+      await selectComponentsForTest(editor, [elementPathInInnards('a7b/b15/b0e')])
+
+      const rootInstance = MetadataUtils.findElementByElementPath(
+        editor.getEditorState().editor.jsxMetadata,
+        elementPathInInnards('a7b'),
+      )
+      if (rootInstance == null) {
+        throw new Error('containerInstance should not be null')
+      }
+
+      const globalContentBoxForRoot = MetadataUtils.getGlobalContentBoxForChildren(rootInstance)
+
+      expect(globalContentBoxForRoot).toEqual({
+        x: 0,
+        y: 0,
+        width: 400,
+        height: 400,
+      })
+
+      const containerInstance = MetadataUtils.findElementByElementPath(
+        editor.getEditorState().editor.jsxMetadata,
+        elementPathInInnards('a7b/b15'),
+      )
+      if (containerInstance == null) {
+        throw new Error('containerInstance should not be null')
+      }
+
+      const globalContentBoxForChildrenOfContainer =
+        MetadataUtils.getGlobalContentBoxForChildren(containerInstance)
+
+      expect(globalContentBoxForChildrenOfContainer).toEqual({
+        x: 140,
+        y: 130,
+        width: 0,
+        height: 150,
+      })
+
+      const childInstance = MetadataUtils.findElementByElementPath(
+        editor.getEditorState().editor.jsxMetadata,
+        elementPathInInnards(`a7b/b15/b0e`),
+      )
+      if (childInstance == null) {
+        throw new Error('childInstance should not be null')
+      }
+
+      const globalContentBoxForChild = MetadataUtils.getGlobalContentBoxForChildren(childInstance)
```

Comment:
may be nitpick, but img cannot have children, so globalContentBoxForChild on the img is probably not the best example use case! for example in the future if we find a better dom API instead of [isElementAContainingBlockForAbsolute](https://github.com/concrete-utopia/utopia/blob/e3cd990b4915af11f736cb3ab6f4ef898f4c2a73/editor/src/components/canvas/dom-walker.ts#L119) then this may become (rightfully!) the globalFrame of the containingBlock parent